app/controllers/api/v1/packages_controller.rb

```diff
@@ -88,17 +94,54 @@ def add_item_to_stockit
     def package_params
       attributes = [:quantity, :length, :width, :height, :notes, :item_id,
         :received_at, :rejected_at, :package_type_id, :state_event, :image_id,
-        :inventory_number, :location_id]
+        :inventory_number, :location_id, :designation_name]
       params.require(:package).permit(attributes)
     end
 
+    def item_attributes
+      if (item = item_params)
+        item["donor_condition_id"] = DonorCondition.find_by(name_en: item["donor_condition_id"]).try(:id)
+        item
+      end
+    end
+
+    def item_params
+      params["package"].require(:item).permit(:donor_condition_id)
+    end
+
+    def save_item_details
+      params["package"]["item"] &&
+      (attributes = item_attributes) &&
+      @package.item.update_attributes(attributes)
```

Comment:
@steveyken purpose to add `attributes` local variable is just to avoid extra method call `item_attributes`
As in `item_attributes`, we look for donor-condition, which might cause extra `find_by` query again.
Please let me know your thoughts, Thanks.